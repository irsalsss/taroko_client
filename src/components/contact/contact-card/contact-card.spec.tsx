import {
  render,
  screen,
  userEvent,
  waitFor,
} from "@/utils/test/react-testing-setup";
import ContactCard from "./contact-card";

describe("ContactCard", () => {
  it("should render contact card successfully", () => {
    const { baseElement } = render(
      <ContactCard
        isFavorite={true}
        onDeleteContact={jest.fn}
        onEditContact={jest.fn}
        onFavoriteContact={jest.fn}
        contact={{
          id: 1,
          firstName: "James",
          lastName: "Bond",
          job: "Software Engineer",
          description: "Cool guy from Land of Dawn",
        }}
      />
    );

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });

  it("should click favorite, edit and delete icon", async () => {
    const onEditContact = jest.fn();
    const onDeletedContact = jest.fn();
    const onFavoriteContact = jest.fn();

    render(
      <ContactCard
        isFavorite={false}
        onDeleteContact={onDeletedContact}
        onEditContact={onEditContact}
        onFavoriteContact={onFavoriteContact}
        contact={{
          id: 1,
          firstName: "James",
          lastName: "Bond",
          job: "Software Engineer",
          description: "Cool guy from Land of Dawn",
        }}
      />
    );

    const editIcon = screen.getByRole("button", { name: "edit-icon" });
    userEvent.click(editIcon);

    const deleteIcon = screen.getByRole("button", { name: "delete-icon" });
    userEvent.click(deleteIcon);

    const favIcon = screen.getByRole("button", { name: "star-icon" });
    userEvent.click(favIcon);

    await waitFor(() => {
      expect(onEditContact).toHaveBeenCalled();
      expect(onDeletedContact).toHaveBeenCalled();
      expect(onFavoriteContact).toHaveBeenCalled();
    });
  });
});
