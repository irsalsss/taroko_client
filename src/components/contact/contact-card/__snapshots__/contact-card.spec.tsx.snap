// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCard should render contact card successfully 1`] = `
<body>
  <div>
    <div
      class="contact-card-container"
    >
      <div
        class="contact-container"
      >
        <button
          class="button-icon"
          role="button"
        >
          <span
            class="sr-only"
          >
            star-filled-icon
          </span>
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.22303 0.665992C7.32551 0.419604 7.67454 0.419604 7.77702 0.665992L9.41343 4.60039C9.45663 4.70426 9.55432 4.77523 9.66645 4.78422L13.914 5.12475C14.18 5.14607 14.2878 5.47802 14.0852 5.65162L10.849 8.42374C10.7636 8.49692 10.7263 8.61176 10.7524 8.72118L11.7411 12.866C11.803 13.1256 11.5206 13.3308 11.2929 13.1917L7.6564 10.9705C7.5604 10.9119 7.43965 10.9119 7.34365 10.9705L3.70718 13.1917C3.47945 13.3308 3.19708 13.1256 3.25899 12.866L4.24769 8.72118C4.2738 8.61176 4.23648 8.49692 4.15105 8.42374L0.914889 5.65162C0.712228 5.47802 0.820086 5.14607 1.08608 5.12475L5.3336 4.78422C5.44573 4.77523 5.54342 4.70426 5.58662 4.60039L7.22303 0.665992Z"
              fill="#e6b400"
            />
          </svg>
        </button>
        <div
          class="contact-other-information-container"
        >
          <span
            class="font-bold capitalize"
          >
            James Bond
          </span>
          <div
            class="contact-other-information"
          >
            <span>
              Job:
            </span>
            <span>
              Software Engineer
            </span>
          </div>
          <div
            class="contact-other-information"
          >
            <span>
              Description:
            </span>
            <span>
              Cool guy from Land of Dawn
            </span>
          </div>
        </div>
      </div>
      <div
        class="contact-action-container"
      >
        <button
          class="button-icon"
          role="button"
        >
          <span
            class="sr-only"
          >
            edit-icon
          </span>
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.1464 1.14645C12.3417 0.951184 12.6583 0.951184 12.8535 1.14645L14.8535 3.14645C15.0488 3.34171 15.0488 3.65829 14.8535 3.85355L10.9109 7.79618C10.8349 7.87218 10.7471 7.93543 10.651 7.9835L6.72359 9.94721C6.53109 10.0435 6.29861 10.0057 6.14643 9.85355C5.99425 9.70137 5.95652 9.46889 6.05277 9.27639L8.01648 5.34897C8.06455 5.25283 8.1278 5.16507 8.2038 5.08907L12.1464 1.14645ZM12.5 2.20711L8.91091 5.79618L7.87266 7.87267L8.12731 8.12732L10.2038 7.08907L13.7929 3.5L12.5 2.20711ZM9.99998 2L8.99998 3H4.9C4.47171 3 4.18056 3.00039 3.95552 3.01877C3.73631 3.03668 3.62421 3.06915 3.54601 3.10899C3.35785 3.20487 3.20487 3.35785 3.10899 3.54601C3.06915 3.62421 3.03669 3.73631 3.01878 3.95552C3.00039 4.18056 3 4.47171 3 4.9V11.1C3 11.5283 3.00039 11.8194 3.01878 12.0445C3.03669 12.2637 3.06915 12.3758 3.10899 12.454C3.20487 12.6422 3.35785 12.7951 3.54601 12.891C3.62421 12.9309 3.73631 12.9633 3.95552 12.9812C4.18056 12.9996 4.47171 13 4.9 13H11.1C11.5283 13 11.8194 12.9996 12.0445 12.9812C12.2637 12.9633 12.3758 12.9309 12.454 12.891C12.6422 12.7951 12.7951 12.6422 12.891 12.454C12.9309 12.3758 12.9633 12.2637 12.9812 12.0445C12.9996 11.8194 13 11.5283 13 11.1V6.99998L14 5.99998V11.1V11.1207C14 11.5231 14 11.8553 13.9779 12.1259C13.9549 12.407 13.9057 12.6653 13.782 12.908C13.5903 13.2843 13.2843 13.5903 12.908 13.782C12.6653 13.9057 12.407 13.9549 12.1259 13.9779C11.8553 14 11.5231 14 11.1207 14H11.1H4.9H4.87934C4.47686 14 4.14468 14 3.87409 13.9779C3.59304 13.9549 3.33469 13.9057 3.09202 13.782C2.7157 13.5903 2.40973 13.2843 2.21799 12.908C2.09434 12.6653 2.04506 12.407 2.0221 12.1259C1.99999 11.8553 1.99999 11.5231 2 11.1207V11.1206V11.1V4.9V4.87935V4.87932V4.87931C1.99999 4.47685 1.99999 4.14468 2.0221 3.87409C2.04506 3.59304 2.09434 3.33469 2.21799 3.09202C2.40973 2.71569 2.7157 2.40973 3.09202 2.21799C3.33469 2.09434 3.59304 2.04506 3.87409 2.0221C4.14468 1.99999 4.47685 1.99999 4.87932 2H4.87935H4.9H9.99998Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="button-icon"
          role="button"
        >
          <span
            class="sr-only"
          >
            delete-icon
          </span>
          <svg
            fill="none"
            height="15"
            viewBox="0 0 15 15"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M5.5 1C5.22386 1 5 1.22386 5 1.5C5 1.77614 5.22386 2 5.5 2H9.5C9.77614 2 10 1.77614 10 1.5C10 1.22386 9.77614 1 9.5 1H5.5ZM3 3.5C3 3.22386 3.22386 3 3.5 3H5H10H11.5C11.7761 3 12 3.22386 12 3.5C12 3.77614 11.7761 4 11.5 4H11V12C11 12.5523 10.5523 13 10 13H5C4.44772 13 4 12.5523 4 12V4L3.5 4C3.22386 4 3 3.77614 3 3.5ZM5 4H10V12H5V4Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</body>
`;
